tests/drivers/build_all/ieee802154/app.overlay

```diff
@@ -0,0 +1,80 @@
+/*
```

Comment:
I'm a bit puzzled here. This seems to be a copy of the overlay file for `native_posix` platform:
https://github.com/zephyrproject-rtos/zephyr/blob/main/tests/drivers/build_all/ieee802154/boards/native_posix.overlay

Why to enforce this overlay for every other platform/configuration?